src/InvertedIndices.jl

```diff
@@ -160,4 +160,7 @@ end
 @inline Base.to_indices(A, I::Tuple{Not{<:AbstractArray{Bool, 1}}}) = to_indices(A, (eachindex(IndexLinear(), A),), I)
 @inline Base.to_indices(A, I::Tuple{Not{<:Union{Array{Bool}, BitArray}}}) = to_indices(A, (eachindex(A),), I)
 
+@inline Base.getindex(nt::NamedTuple, I::InvertedIndex{Symbol}) = nt[filter(!=(I.skip), keys(nt))]
+@inline Base.getindex(nt::NamedTuple, I::InvertedIndex{NTuple{N, Symbol}}) where {N} = nt[filter(∉(I.skip), keys(nt))]
```

Comment:
This will only be compatible with 1.7, as you noted in the tests. To make this work on earlier versions, you could make the method bodies, respectively,
```julia
(; (k for k in keys(nt) if k != I.skip)...)
(; (k for k in keys(nt) if !(k in I.skip))...)
```
(For some reason, GitHub isn't allowing me to insert a suggestion :thinking:)

Reply:
Maybe we should add these two methods only if we are on Julia 1.7 or later and clearly document it?

Reply:
Come to think of it, the upstream `getindex` methods could be added to Compat if they haven't already been.